docker/profiles/docker-compose.gms.yml

```diff
@@ -113,7 +113,7 @@ x-datahub-gms-service-dev: &datahub-gms-service-dev
   environment: &datahub-gms-dev-env
     <<: [*datahub-dev-telemetry-env, *datahub-gms-env]
     SKIP_ELASTICSEARCH_CHECK: false
-    METADATA_SERVICE_AUTH_ENABLED: false
+    METADATA_SERVICE_AUTH_ENABLED: true
```

Comment:
Instead of enabling this here, we should look at enabling it in the [application.yml](https://github.com/datahub-project/datahub/blob/master/metadata-service/configuration/src/main/resources/application.yml#L7)

Leave this change as you have it but also add the following to the non-dev instance.
`METADATA_SERVICE_AUTH_ENABLED: false`

The result is that the application by default has api auth enabled. It is disabled for quickstart (non-dev) and then reenabled for (dev). This implements a default deny but then still allows CI and quickstart users to avoid dealing with it (although I believe most of the CI does exercise using credentials)

Reply:
Is this what you intended https://github.com/datahub-project/datahub/pull/9793/commits/1e61be6f915f102e71d4038b4405ac681a67fbfd?

Reply:
I should probably also remove this line from `x-datahub-gms-service-dev` now given we have enabled in `application.yml`, correct?

Reply:
Caused many unautheticated errors in the CI so removing this change in this PR. Will just focus on the new endpoint here